python/cog.py

```diff
@@ -117,11 +117,58 @@ async def slash_search(self, ctx: SlashContext, **kwargs):
         if isinstance(files, str):
             await ctx.send(content=files, hidden=True)
             return
-        author = ctx
+
+        recipient = ctx
         if kwargs.get("dm"):
-            author = ctx.author
+            recipient = ctx.author
             await ctx.send("DM'ing your files...", hidden=True)
-        await self.send_files_as_message(author, files)
+
+        return (recipient, files)
+
+    @cog_ext.cog_slash(
+        name="search",
+        description="Search for files.",
+        options=search_options,
+        guild_ids=guild_ids if getattr(CONFIG, "DB_NAME", "production") == "testing" else []
+    )
+    async def slash_search(self, ctx: SlashContext, **kwargs):
+        """
+        Responds to `/search`. Tries to display docs related to a query from ElasticSearch.
+
+        Args:
+            ctx: The SlashContext from which the command originated
+            filename: A str of the filename to query for.
+            DM: A bool for whether to dm the author the results.
+        """
+        recipient, files = await self.locate(ctx, **kwargs)
+        await self.send_files_as_message(recipient, files)
+
+    @cog_ext.cog_slash(
+        name="export",
+        description="Get a Python export script to download the files returned in a search to your computer.",
+        options=search_options,
+        guild_ids=guild_ids if getattr(CONFIG, "DB_NAME", "production") == "testing" else []
+    )
+    async def slash_export(self, ctx: SlashContext, **kwargs):
+        """
+        Responds to `/export`. Tries get a download script for all files related to a query from ElasticSearch.
+
+        Args:
+            ctx: The SlashContext from which the command originated.
+            filename: A str of the filename to query for, if any.
+            DM: A bool for whether to dm the author the results.
+        """
+        recipient, files = await self.locate(ctx, **kwargs)
+        channels = {}
+        for c in ctx.guild.channels:
+            channels[str(c.id)] = c.name
+        export_name = re.sub(r"[^A-Za-z0-9'\-\_ ]", "", ctx.guild.name)
```

Comment:
When you get a chance, could you throw some whacky stuff in the server name and test this out?